polar-wasm-api/src/errors.rs

```diff
@@ -49,6 +48,8 @@ fn kind(err: &PolarError) -> String {
         Validation(InvalidRuleType { .. }) => "ValidationError::InvalidRuleType",
         Validation(ResourceBlock { .. }) => "ValidationError::ResourceBlock",
         Validation(UndefinedRule { .. }) => "ValidationError::UndefinedRule",
+        Validation(SingletonVariable { .. }) => "ValidationError::SingletonVariable",
+        Validation(UnregisteredConstant { .. }) => "ValidationError::UnregisteredConstant",
```

Comment:
Any changes required in JS & other libraries to handle this new error variant?

Reply:
Go types will need updating.

Reply:
> Go types will need updating.

7bd53f89

Reply:
> Any changes required in JS & other libraries to handle this new error variant?

- JS: none — all errors crossing the WASM boundary from the core are regular old `Error` instances that have the stringified variant name from the core attached as the `name` field.
- Ruby: none — the code for handling errors received across the FFI boundary is fairly out-of-sync with the core, but not in a way that will currently break anything. All four error variants (parse, runtime, validation, operational) have fallback translations that will handle new errors fine. Users won't be able to specifically pattern match on the new error type (just the generic variant, e.g., `ValidationError` in this case), but that's something I'd like to address in another error cleanup follow-up.
- Python: none — same as Ruby, including being out-of-sync with the core.
- Go: none — errors have been generated (see above comment).
- Rust: none.
- Java: none — same as Python & Ruby, including being out-of-sync with the core.